renderdoc/driver/gl/gl_replay_egl.cpp

```diff
@@ -94,6 +222,159 @@ ReplayCreateStatus GL_CreateReplayDevice(const char *logfile, IReplayDriver **dr
 {
   RDCDEBUG("Creating an OpenGL ES replay device");
 
-  // TODO
-  return eReplayCreate_APIInitFailed;
+  // Query the required EGL functions
+  if(eglCreateContextProc == NULL)
+  {
+    eglGetProcAddressProc = (PFN_eglGetProcAddress)dlsym(RTLD_NEXT, "eglGetProcAddress");
+    eglChooseConfigProc = (PFN_eglChooseConfig)dlsym(RTLD_NEXT, "eglChooseConfig");
+    eglInitializeProc = (PFN_eglInitialize)dlsym(RTLD_NEXT, "eglInitialize");
+    eglBindAPIProc = (PFN_eglBindAPI)dlsym(RTLD_NEXT, "eglBindAPI");
+    eglGetDisplayProc = (PFN_eglGetDisplay)dlsym(RTLD_NEXT, "eglGetDisplay");
+    eglCreateContextProc = (PFN_eglCreateContext)dlsym(RTLD_NEXT, "eglCreateContext");
+    eglMakeCurrentProc = (PFN_eglMakeCurrent)dlsym(RTLD_NEXT, "eglMakeCurrent");
+    eglSwapBuffersProc = (PFN_eglSwapBuffers)dlsym(RTLD_NEXT, "eglSwapBuffers");
+    eglDestroyContextProc = (PFN_eglDestroyContext)dlsym(RTLD_NEXT, "eglDestroyContext");
+    eglDestroySurfaceProc = (PFN_eglDestroySurface)dlsym(RTLD_NEXT, "eglDestroySurface");
+    eglQuerySurfaceProc = (PFN_eglQuerySurface)dlsym(RTLD_NEXT, "eglQuerySurface");
+    eglCreatePbufferSurfaceProc =
+        (PFN_eglCreatePbufferSurface)dlsym(RTLD_NEXT, "eglCreatePbufferSurface");
+    eglCreateWindowSurfaceProc =
+        (PFN_eglCreateWindowSurface)dlsym(RTLD_NEXT, "eglCreateWindowSurface");
+
+    if(eglGetProcAddressProc == NULL || eglBindAPIProc == NULL || eglGetDisplayProc == NULL ||
+       eglCreateContextProc == NULL || eglMakeCurrentProc == NULL || eglSwapBuffersProc == NULL ||
+       eglDestroyContextProc == NULL || eglDestroySurfaceProc == NULL ||
+       eglQuerySurfaceProc == NULL || eglCreatePbufferSurfaceProc == NULL ||
+       eglCreateWindowSurfaceProc == NULL || eglChooseConfigProc == NULL)
+    {
+      RDCERR(
+          "Couldn't find required function addresses, eglGetProcAddress eglCreateContext"
+          "eglSwapBuffers (etc.)");
+      return eReplayCreate_APIInitFailed;
+    }
+  }
+
+  GLInitParams initParams;
+  RDCDriver driverType = RDC_OpenGL;
```

Comment:
This raises a question - do we want to have captures be cross-compatible between GLES and Desktop GL? i.e. being able to capture on one and replay on another.

I think it'd be a useful feature, I'm just not sure if it's practical. You'd be able to say better than me whether concepts can be translated back and forth, but I'd guess it's common to use extensions that won't have any equivalent on desktop.

If they're going to be distinct (ie. GLES can only be replayed on GLES devices) then you should add a new driver type and set it here, with the name set to OpenGL ES.